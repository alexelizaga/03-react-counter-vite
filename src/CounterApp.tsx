import { MouseEvent, useState } from 'react';

interface CounterAppProps {
    value: number;
}

export const CounterApp = ({ value }: CounterAppProps) => {

    const [ counter, setCounter ] = useState(value)

    const handleAdd = ( event: MouseEvent ) => {
        setCounter( prev => prev + 1 );
    }

    const handleSubstract = ( event: MouseEvent ) => {
        setCounter( prev => prev - 1 );
    }

    const handleReset = ( event: MouseEvent ) => {
        setCounter( value );
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>

            <button onClick={ handleAdd }>+1</button>
            <button onClick={ handleSubstract }>-1</button>
            <button onClick={ handleReset }>Reset</button>
        </>
    )
}
