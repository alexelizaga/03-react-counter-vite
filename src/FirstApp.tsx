interface FirstAppProps {
    title?: string;
    subtitle?: string;
    name?: string;
}

export const FirstApp = ( { title = 'No title', subtitle = 'No subtitle', name = 'Alex' }: FirstAppProps ) => {

    return (
        <>
            <h1>{ title }</h1>
            <p>{ subtitle }</p>
            <p>{ name }</p>
        </>
    )
}

