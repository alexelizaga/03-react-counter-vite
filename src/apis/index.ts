export { default as giphyApi } from './giphyApi';