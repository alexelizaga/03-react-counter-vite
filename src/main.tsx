import React from 'react';
import ReactDOM from 'react-dom/client';

import { FirstApp } from './FirstApp';

import './styles.css';


ReactDOM.createRoot(document.getElementById('root')!).render(
  <React.StrictMode>
    <FirstApp title='Alex' subtitle='I am a subtitle' />
  </React.StrictMode>
)
