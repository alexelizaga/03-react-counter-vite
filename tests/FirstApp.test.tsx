/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { FirstApp } from '../src/FirstApp';

describe('Test <FirstApp />', () => {
    
    test('should match snapshot v1', () => {
        
        const component = renderer.create(
            <FirstApp title='' />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    });

    test('should match snapshot v2', () => {
        
        const title = 'Hello, I am Goku';
        render( <FirstApp title={ title } /> );

    });

});