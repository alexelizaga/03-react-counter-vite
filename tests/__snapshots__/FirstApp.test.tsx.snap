// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <FirstApp /> should match snapshot 1`] = `
Array [
  <h1 />,
  <p>
    No subtitle
  </p>,
  <p>
    Alex
  </p>,
]
`;

exports[`Test <FirstApp /> should match snapshot v1 1`] = `
Array [
  <h1 />,
  <p>
    No subtitle
  </p>,
  <p>
    Alex
  </p>,
]
`;
